conan_provider.cmake

```diff
@@ -247,6 +247,25 @@ function(detect_host_profile output_file)
 
     string(APPEND PROFILE "[conf]\n")
     string(APPEND PROFILE "tools.cmake.cmaketoolchain:generator=${CMAKE_GENERATOR}\n")
+
+    # propagate compilers via profile
+    set(_conan_c_compiler "")
```

Comment:
I'll move this to its own helper ``define_compiler_executables_conf`` or the like.